src/main/java/com/mongodb/hibernate/internal/translate/ModelMutationMqlTranslator.java

```diff
@@ -40,7 +41,13 @@ public O translate(@Nullable JdbcParameterBindings jdbcParameterBindings, QueryO
         assertNull(jdbcParameterBindings);
         checkQueryOptionsSupportability(queryOptions);
 
-        var mutationCommand = acceptAndYield(tableMutation, COLLECTION_MUTATION);
-        return tableMutation.createMutationOperation(renderMongoAstNode(mutationCommand), getParameterBinders());
+        String sql;
+        if ((TableMutation<?>) tableMutation instanceof TableUpdateNoSet) {
+            sql = "";
+        } else {
+            var mutationCommand = acceptAndYield(tableMutation, COLLECTION_MUTATION);
+            sql = renderMongoAstNode(mutationCommand);
+        }
+        return tableMutation.createMutationOperation(sql, getParameterBinders());
```

Comment:
Turned out, Hibernate ORM has an implementation of `TableMutation`, whose `accept` method does nothing. Passing an instance of `TableUpdateNoSet` to `acceptAndYield` results in `AssertionError`, because the `accept` method does nothing, which means, `AstVisitorValueHolder.yield` is not called, which we don't expect.

Without this change, the new `nestedEmptyValue` test fails.